src/Components/GenreSelector/GenreSelector.js

```diff
@@ -0,0 +1,39 @@
+import React from 'react';
+import classNames from 'classnames';
+const title = 'Genre Selector';
+
+class GenreSelector extends React.Component {
+    constructor(props) {
+        super(props);
+        this.state = {
+            selected: parseInt(props.defaultSelectedGenre),
+            genreList: props.genreList
+        };
+    }
+
+    selectGenre(genre){
+        const _self = this;
```

Comment:
Just for your information
You can rewrite it in the next way:

you can put a function inside the this.setState and that function will have the current state in the params

```
    selectGenre(genre){
        this.setState(function (prevState) {
            return {
                ...prevState,
                selected: genre.id
            }
        });
        this.props.onSelectGenre(genre)
    };

```